src/Update.cpp

```diff
@@ -610,12 +610,20 @@ void DoDetectedCase(int ai, double t, unsigned short int ts, int tn)
 		}
 	if (P.DoHouseholds)
 	{
-		if ((!P.DoMassVacc) && (t >= P.VaccTimeStart) && (State.cumV < P.VaccMaxCourses))
-			if ((t < P.VaccTimeStart + P.VaccHouseholdsDuration) && ((P.VaccPropCaseHouseholds == 1) || (ranf_mt(tn) < P.VaccPropCaseHouseholds)))
+		if ((!P.DoMassVacc) && (t >= P.VaccTimeStart))
+		{
+			// DoVacc is going to test that `State.cumV < P.VaccMaxCourses` itself before
+			// incrementing State.cumV, but by checking here too we can avoid a lot of
+			// wasted effort
+			bool cumV_OK;
+#pragma omp critical (state_cumV)
+			cumV_OK = State.cumV < P.VaccMaxCourses;
```

Comment:
It is not obvious to me whether this critical section lasts for the rest of the block or just the next statement.  I believe we want just the statement - so can we put it in a block just to make it obvious please?